server/src/main/java/org/opensearch/cluster/decommission/DecommissionStatus.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+/**
+ * An enumeration of the states during decommissioning and recommissioning.
+ */
+public enum DecommissionStatus {
+    /**
+     * Decommission process is initiated
+     */
+    DECOMMISSION_INIT("decommission_init"),
+    /**
+     * Decommission process has started, decommissioned nodes should be weighed away
+     */
+    DECOMMISSION_IN_PROGRESS("decommission_in_progress"),
+    /**
+     * Decommissioning awareness attribute completed
+     */
+    DECOMMISSION_SUCCESSFUL("decommission_successful"),
+    /**
+     * Decommission request failed
+     */
+    DECOMMISSION_FAILED("decommission_failed");
```

Comment:
Do we want to include voting config change as a state transition or looking at voting config should tell us that

Reply:
Didn't see a use case where we need to add exclusion status in metadata as we know whenever the to-be-decommissioned leader abdicates only then we initiate the decommissioning and if it doesn't we can certainly say that request for decommissioning will not move forward. And if we see a problem with INIT we clear and exclusion and exit

Reply:
Update - We start off with INIT and then proceed to exclude the leader from voting config. Once that is done status is updated to IN_PROGRESS and proceeds with removing nodes